myia/anf_ir.py

```diff
@@ -42,6 +42,25 @@ def __init__(self) -> None:
         self.return_: Apply = None
         self.debug = GraphDebug()
 
+    def __str__(self) -> str:
```

Comment:
I'm a bit iffy on having this kind of string representation for graphs. It's tricky to do it properly, and it's even trickier to do it in a way that doesn't look like garbage for non-trivial graphs. For example, for:

```python
def f(x):
    a = x + 1
    b = x + 2
    return a + b
```

`str(parse(f))` currently gives me `f(x) -> add(a, b)` which is broken. It also busts the stack for some recursive graphs.

It's not trivial to fix either, since you only want to give `a`'s definition once even if it is used more than once, you may need to create temporary names for nameless nodes that are reused, and so on. It's work, basically. Graphical representations are easier to implement and more usable so I'm not sure it's worth bothering with textual ones at all. I'm working on porting the colourful graphs I had on master, for what it's worth.

Reply:
I agree that this quickly became too complicated. Even if we keep textual representations, I think they should be factored out into separate debugging tools.

That said, it would be useful to have a limited representation for when you're inspecting objects in `pdb` e.g. maybe something like `Graph(args=['x'], return=Node(inputs=['add', 'a', 'b']))` instead of `<myia.Graph at 0x1095da748>`?

Reply:
Sure, we can have something like that, plus the graph's name if one is available.